@page "/AgregarEvento"
@attribute [Authorize]
@using sistemaReservas.Shared
@using System.Net.Http
@using System.Net.Http.Json
@using System.Threading.Tasks
@using System.ComponentModel.DataAnnotations
@inject HttpClient Http


<PageTitle>Agregar Evento</PageTitle>
<h1>Agregar Evento</h1>

<EditForm Model="@agregarEventoParameters" OnValidSubmit="@OnSubmit">
	<DataAnnotationsValidator />
    <HxInputText LabelType="LabelType.Floating" Label="Nombre" @bind-Value="agregarEventoParameters.Nombre" CssClass="my-3"/>

	<HxInputDate LabelType="LabelType.Floating" Label="Fecha" CalendarIcon="BootstrapIcon.CalendarEvent" @bind-Value="agregarEventoParameters.Fecha" MinDate="@DateTime.Today" CssClass="my-3"/>

	
	<HxSubmit Color="ThemeColor.Primary" Icon="BootstrapIcon.CalendarPlus">Agregar Evento</HxSubmit>

	@if (exito == true)
		{			
			<hr>
			<HxAlert Color="ThemeColor.Success" Dismissible="true"><HxIcon Icon="@BootstrapIcon.CheckCircleFill" /> ¡El evento se cargó correctamente!</HxAlert>			
			agregarEventoParameters.Nombre = null;
			agregarEventoParameters.Fecha = null;			
			request = null;
			
		}

	@if (exito == false)
	{
		<hr>
		<HxAlert Color="ThemeColor.Danger"><HxIcon Icon="@BootstrapIcon.ExclamationTriangleFill" />
			<h4 class="alert-heading"> Ocurrió un error</h4>
			<p>Por favor, revise los datos ingresados e intente nuevamente</p>
			<hr>
			<p class="mb-0">Si el error persiste, no dude en contactar al Soporte Técnico.</p>
			
			if(error != null) 
			{
				<hr>
				<p class="mb-0">@error</p>
			} 
		</HxAlert>		
	}
</EditForm>

@code {
	protected AgregarEventoParameters agregarEventoParameters = new AgregarEventoParameters();
	string? error { get; set; } = null;
	private HttpResponseMessage? request;
	private bool? exito = null;

	private async Task OnSubmit()
	{
		error = null;
		exito = null;
		try		
		{						
			request = await Http.PostAsJsonAsync("Api/Eventos", agregarEventoParameters);
			if (request.IsSuccessStatusCode)
			{ 
				exito = true;
			}
			else
			{
				exito = false;
			}

		}
		catch (Exception ex)
		{
			exito = false;
			error = ex.Message;
        }
		

    }


}
